memcache/memcache.go

```diff
@@ -134,18 +134,25 @@ func NewDiscoveryClient(discoveryAddress string, pollingDuration time.Duration)
 	if pollingDuration.Seconds() < 1.0 {
 		return nil, ErrInvalidPollingDuration
 	}
-	return newDiscoveryClient(discoveryAddress, pollingDuration)
+	return newDiscoveryClient(discoveryAddress, new(ServerList), pollingDuration)
+}
+
+func NewDynamicRendezvousClient(discoveryAddress string, pollingDuration time.Duration) (*Client, error) {
+	// Validate pollingDuration
+	if pollingDuration.Seconds() < 1.0 {
+		return nil, ErrInvalidPollingDuration
+	}
```

Comment:
you don't want to move this check to `newDiscoveryClient`?

Reply:
yeah good call!